### ISSUE TITLE
DISABLED test_call_method_on_rref (__main__.TensorPipeAgentRpcTestWithSpawn)

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#38757]
https://app.circleci.com/pipelines/github/pytorch/pytorch/171959/workflows/3211d854-a984-469e-8984-602211e6a3d6/jobs/5528991/steps

```
May 20 01:59:05 ======================================================================
May 20 01:59:05 FAIL [1.215s]: test_call_method_on_rref (__main__.TensorPipeAgentRpcTestWithSpawn)
May 20 01:59:05 ----------------------------------------------------------------------
May 20 01:59:05 Traceback (most recent call last):
May 20 01:59:05   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 204, in wrapper
May 20 01:59:05     self._join_processes(fn)
May 20 01:59:05   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 306, in _join_processes
May 20 01:59:05     self._check_return_codes(elapsed_time)
May 20 01:59:05   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 349, in _check_return_codes
May 20 01:59:05     i, first_process.exitcode, p.exitcode
May 20 01:59:05   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1116, in assertEqual
May 20 01:59:05     self.assertTrue(result, msg=message)
May 20 01:59:05 AssertionError: False is not true : Expect process 1 exit code to match Process 0 exit code of -6, but got 0
```

```
May 20 01:56:59   test_call_method_on_rref (__main__.TensorPipeAgentRpcTestWithSpawn) ... [W tensorpipe_agent.cpp:222] RPC agent is being closed. Skip sending rpc response
May 20 01:56:59 [W tensorpipe_agent.cpp:258] Server read message: EOF: end of file
May 20 01:56:59 [W tensorpipe_agent.cpp:258] Server read message: EOF: end of file
May 20 01:56:59 [W tensorpipe_agent.cpp:383] Read response error: EOF: end of file
May 20 01:56:59 terminate called after throwing an instance of 'std::runtime_error'
May 20 01:56:59   what():  EOF: end of file
```

cc @ezyang @gchanan @zou3519 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528 @jjlilley @osalpekar @jiayisuse @lw @beauby

**Comment by mrshenli** [comment]
one more occurrence: https://app.circleci.com/pipelines/github/pytorch/pytorch/172015/workflows/a8f2bda7-4d66-4df7-aba9-3662cc200910/jobs/5530054/steps

**Comment by lw** [comment]
All tests should now be fixed as part of the stack ending in #39441.
